File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task166_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.*;

public class Main {
    public static int[] posNegSort(int[] arr) {
        List<Integer> pos = new ArrayList<>();
        for (int x : arr) {
            if (x > 0) {
                pos.add(x);
            }
        }
        Collections.sort(pos);
        int[] res = new int[arr.length];
        int pos_index = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > 0) {
                res[i] = pos.get(pos_index);
                pos_index++;
            } else {
                res[i] = arr[i];
            }
        }
        return res;
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task166_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <vector>
#include <algorithm>

std::vector<int> posNegSort(std::vector<int> arr) {
    std::vector<int> pos;
    for (int x : arr) {
        if (x > 0) {
            pos.push_back(x);
        }
    }
    std::sort(pos.begin(), pos.end());
    int pos_index = 0;
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] > 0) {
            arr[i] = pos[pos_index];
            pos_index++;
        }
    }
    return arr;
}

int main() {
    std::vector<int> arr = {6, 3, -2, 5, -8, 2, -2};
    arr = posNegSort(arr);
    for (int x : arr) {
        std::cout << x << " ";
    }
    return 0;
}